ssh/tailssh/tailssh.go

```diff
@@ -390,6 +397,22 @@ func (c *conn) ServerConfig(ctx ssh.Context) *gossh.ServerConfig {
 
 			return perms, nil
 		},
+		PasswordCallback: func(cm gossh.ConnMetadata, password []byte) (*gossh.Permissions, error) {
+			c.logf("ZZZZ PasswordCallback")
+			// Some clients don't request 'none' authentication. Instead, they
```

Comment:
Yes they do. That's how the protocol goes. They're not handling the "that's okay!" in response to that.

Reply:
All that said, I've not yet found a client that starts with password, so I took that back out for now.

Reply:
I've added an [integration test](https://github.com/tailscale/tailscale/pull/14970/files#diff-1db48620d325c9395aedd1e083957444227ec124a0d80968cc00b87348bf8876R417) that uses `paramiko`. Without the fix, this test fails.